src/Storages/ExternalStream/Kafka/Consumer.cpp

```diff
@@ -51,6 +49,9 @@ std::vector<Int64> Consumer::getOffsetsForTimestamps(const std::string & topic,
 
 void Consumer::startConsume(Topic & topic, Int32 parition, Int64 offset)
 {
+    if (!started.test_and_set())
```

Comment:
are we expecting startConsume  being called multiple times ? may be we will need do

```
if (started.test_and_set())
    return;

...
```